src/Workspaces/Core/Portable/Shared/Extensions/ContextQuery/AbstractSyntaxContext.cs

```diff
@@ -87,6 +93,12 @@ private ISet<INamedTypeSymbol> ComputeOuterTypes(CancellationToken cancellationT
             return SpecializedCollections.EmptySet<INamedTypeSymbol>();
         }
 
+        protected static IEnumerable<ITypeSymbol> ComputeInferredTypes(Workspace workspace, SemanticModel semanticModel, int position, CancellationToken cancellationToken)
```

Comment:
I wound up doing this because the keyword recommender tests don't have a workspace or composition, so I needed a test constructor create an inference service, call it, and pass that to the real constructor. Open to better ways to untangle this :smile: